### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by kleisauke**
See the individual commits for more details. Supersedes: #3.

(I've checked the "Allow edits by maintainers" box, so feel free to commit on this branch)

### CODE REVIEW

File: src/wasm32/ffi.c [hunk 1]
```diff
@@ -344,9 +344,9 @@ ffi_call, (ffi_cif * cif, ffi_fp fn, void *rvalue, void **avalue),
 
 EM_JS_MACROS(void *, ffi_closure_alloc_helper, (size_t size, void **code), {
   var closure = _malloc(size);
-  var func_ptr = getEmptyTableSlot();
-  DEREF_U32(code, 0) = func_ptr;
-  CLOSURE__wrapper(closure) = func_ptr;
+  var index = getEmptyTableSlot();
+  DEREF_U32(code, 0) = index;
+  CLOSURE__wrapper(closure) = index;
```
Comment (hoodmane) [inline src/wasm32/ffi.c:343]: This commit just changes the name of the variable? I sure hope this didn't fix any bugs. Presumably the actual changes here went away after a rebase...

File: testsuite/libffi.closures/huge_struct.c [hunk 2]
```diff
@@ -5,7 +5,7 @@
 	Originator:		Blake Chaffin	6/18/2007
 */
 
-/* { dg-do run { xfail strongarm*-*-* xscale*-*-* } } */
+/* { dg-do run { xfail strongarm*-*-* xscale*-*-* wasm32*-*-* } } */
```
Comment (hoodmane) [inline testsuite/libffi.closures/huge_struct.c:8]: What about `libffi.closures`tests `huge_struct.c` and `closure_loc_fn_0.c`? These should fail too.

File: testsuite/libffi.closures/cls_24byte.c [hunk 3]
```diff
@@ -118,7 +118,7 @@ int main (void)
   args_dbl[4] = NULL;
 
   ffi_call(&cif, FFI_FN(cls_struct_24byte_fn), &res_dbl, args_dbl);
-  /* { dg-output "9 2 6 5 1 2 3 7 4 5 7 9 8 6 1 4: 22 15 17 25" } */
+  /* { dg-output "9 2 6 5 1 2 3 7 4 5 7 9 8 6 1 9: 22 15 17 25" } */
```
Comment (hoodmane) [inline testsuite/libffi.closures/cls_24byte.c:121]: I'm pretty sure that should correctly be a 4:
```
  struct cls_struct_24byte e_dbl = { 9.0, 2.0, 6, 5.0 };
  struct cls_struct_24byte f_dbl = { 1.0, 2.0, 3, 7.0 };
  struct cls_struct_24byte g_dbl = { 4.0, 5.0, 7, 9.0 };
  struct cls_struct_24byte h_dbl = { 8.0, 6.0, 1, 4.0 };
```
So:
9 2 6 5 | 1 2 3 7 | 4 5 7 9 | 8 6 1 **9**
should be
9 2 6 5 | 1 2 3 7 | 4 5 7 9 | 8 6 1 **4**
kinda surprises me that nobody noticed this yet upstream...

**Comment by hoodmane** [comment]
Thanks!

File: src/wasm32/ffi.c (hunk 1 above)
Comment (kleisauke) [reply src/wasm32/ffi.c:343]: It was related to this error:
```
exception thrown: TypeError: Cannot read property 'delete' of undefined,TypeError: Cannot read property 'delete' of undefined
    at removeFunction (/root/repo/testsuite/test-callback:2:5150)
    at ffi_closure_free_helper (/root/repo/testsuite/test-callback:2:12958)
```

Which could be fixed by swapping `removeFunction` call with `freeTableIndexes.push(index)`. See:
https://github.com/emscripten-core/emscripten/blob/4d864df0a57024d667e8db171aab2c3385d04c30/src/runtime_functions.js#L168-L171

Hence, I renamed `func_ptr` variable to `index`.

File: testsuite/libffi.closures/huge_struct.c (hunk 2 above)
Comment (kleisauke) [reply testsuite/libffi.closures/huge_struct.c:8]: Commit https://github.com/hoodmane/libffi-emscripten/pull/4/commits/254af96bc0231077bc2947be5090b6b343ed44ab changed those tests to `xfail` ones.

File: testsuite/libffi.closures/cls_24byte.c (hunk 3 above)
Comment (kleisauke) [reply testsuite/libffi.closures/cls_24byte.c:121]: Oh huh, curious. Perhaps upstream's CI failures were related to this?

Sorry about the this change, I had reverted the entire test suite to see which `dg-output`/`printf` lines had changed.

**Comment by kleisauke** [comment]
No problem, glad I could help!

File: testsuite/libffi.closures/cls_24byte.c (hunk 3 above)
Comment (hoodmane) [reply testsuite/libffi.closures/cls_24byte.c:121]: See https://github.com/libffi/libffi/pull/652

**Pull request merged**